content/blog/modern-text-features/index.Rmd

```diff
@@ -0,0 +1,370 @@
+
+```{r, include=FALSE}
+preview_devices <- function(p, title, width = 2, height = 1) {
+  os <- "macOS"
+  quartz_file <- paste0("figs/", title, "_", os, "_", "quartz.png")
+  if (!dir.exists("figs")) dir.create("figs")
+  png(quartz_file, width, height, units = 'in', res = 300, type = "quartz")
+  plot(
+    p + 
+      ggtitle(paste0("Quartz device (", os, ")")) + 
+      theme(plot.title = element_text(size = 8, hjust = 0.5), plot.title.position = 'plot')
+  )
+  dev.off()
+  cairo_file <- paste0("figs/", title, "_", os, "_", "cairo.png")
+  png(cairo_file, width, height, units = 'in', res = 300, type = "cairo")
+  plot(
+    p + 
+      ggtitle(paste0("Cairo device (", os, ")")) + 
+      theme(plot.title = element_text(size = 8, hjust = 0.5), plot.title.position = 'plot')
+  )
+  dev.off()
+  ragg_file <- paste0("figs/", title, "_", os, "_", "ragg.png")
+  ragg::agg_png(ragg_file, width, height, units = 'in', res = 300)
+  plot(
+    p + 
+      ggtitle(paste0("Ragg device (", os, ")")) + 
+      theme(plot.title = element_text(size = 8, hjust = 0.5), plot.title.position = 'plot',
+            plot.background = element_rect('honeydew', NA))
+  )
+  invisible(dev.off())
+}
+get_results <- function(name) {
+  files <- sort(list.files("figs", pattern = name))
+  files <- files[order(grepl('ragg', files), decreasing = TRUE)]
+  files <- c(
+    grep("macOS", files, value = TRUE),
+    grep("Windows", files, value = TRUE),
+    grep("Linux", files, value = TRUE)
+  )
+  files <- c(files[c(T, F, F)], files[c(F, T, F)], files[c(F, F, T)])
+  cat(paste(paste0("<img src=\"figs/", files, "\" width=\"33%\" style=\"display: inline;\">"), collapse = ""))
+}
+theme_set(theme_void() + theme(panel.background = element_rect('gray95', NA), plot.margin = margin(0, 2, 2, 2)))
+```
+
+## Advanced script support
+
+English, being the lingua franca of programming, has generally dominated everything related to text within programming, ranging from encoding to rendering.
+Because of this, the Latin script, which is used in most of the western world, has been the best (or often only) supported script in many text rendering pipelines.
+This has also been true in the R world where the build-in graphic devices has struggled to display scripts that differed from the standard Latin layout (the Cairo devices on Linux being the exception as we will see).
+This is not a jab at the provided graphic devices.
+ragg has handled text rendering in the exact same way up until now and the work that has gone into changing it has not been trivial.
+Technicalities aside, it is about time (overdue, actually) that the graphic system in R begins to support the display of non-Latin script and becomes more inclusive in terms of which languages can be used.
+It is thus with great joy that I can finally support it in ragg.
+
+### Right-to-Left scripts
+
+To start off we will look at a sample of different scripts that poses a problem due to their direction.
```

Comment:
```suggestion
To start off we will look at a sample of different scripts (Arabic, Hebrew, and Sindhi) that pose a challenge because they are written from right to left:
```